doc/man3/OSSL_HTTP_transfer.pod

```diff
@@ -190,13 +190,20 @@ Since this function is typically called by applications such as
 L<openssl-s_client(1)> it uses the B<bio_err> and B<prog> parameters (unless
 NULL) to print additional diagnostic information in a user-oriented way.
 
-OSSL_HTTP_parse_url() parses its input string B<url> as a URL and splits it up
+OSSL_HTTP_parse_url() parses its input string B<url> as a URL
+of the form C<[http[s]://]address[:port][/path]> and splits it up
 into host, port and path components and a flag whether it begins with 'https'.
-The host component may be a DNS name or an IPv4 or an IPv6 address.
+The host component may be a DNS name or an IPv4 address
+or an IPv6 address enclosed in C<[> and C<]>.
 The port component is optional and defaults to "443" for HTTPS, else "80".
+If the B<pport_num> argument is NULL the port specification
+can be in mnemonic form such as "http" like with L<BIO_set_conn_port(3)>, else
+it must be in numerical form and its integer value is assigned to B<*pport_num>.
 The path component is also optional and defaults to "/".
-As far as the result pointer arguments are not NULL it assigns via
-them copies of the respective string components.
+As far as the result pointer arguments are not NULL the function assigns
+via them the respective url components.
+For string values these are non-NULL pointers to copies of the strings
```

Comment:
'For strings ...' should be collapsed down and merged into the 'The strings merged this way' part.

Reply:
I've re-arranged the sentences to:
```
A path string returned is guaranteed to begin with a C</> character.
On success the function assigns via each non-NULL result pointer argument
the respective url component. On error, string results are assigned to NULL,
else they are guaranteed to be non-NULL pointers to copies of the strings.
It is the reponsibility of the caller to free them using L<OPENSSL_free(3)>.
```
which should read better now :)

Reply:
What about something like this - that still hurts my brain..

If the parameters I<phost>, I<pport> and I<ppath> are non-NULL then the returned strings must be freed by the caller using L<OPENSSL_free(3)>. If an error occurs then the returned strings will be NULL.
If a path string I<pppath> is returned it is guaranteed to begin with a C</> character.

Reply:
Your text suggestion does not include the statement that on success the strings returned as guaranteed to be non-NULL.
Yet I see your point to better explicitly give the parameter names - done this, and the section now reads as:
```
On success the function assigns via each non-NULL result pointer argument
I<phost>, I<pport>, I<pport_num>, I<ppath>, and I<pssl>
the respective url component.
On error, B<*phost>, B<*pport>, and B<*ppath> are assigned to NULL,
else they are guaranteed to contain non-NULL string pointers.
It is the reponsibility of the caller to free them using L<OPENSSL_free(3)>.
A string returned via B<*ppath> is guaranteed to begin with a C</> character.
```
On this occasion I replaced in the file all parameter references `B<...>` by `I<...>`.